projects/storefrontlib/src/cms-components/product/carousel/product-carousel/product-carousel.component.ts

```diff
@@ -29,11 +29,12 @@ export class ProductCarouselComponent {
   );
 
   /**
-   * Obervable that holds an Array of Observables. This is done, so that
+   * Observable that holds an Array of Observables. This is done, so that
    * the component UI could consider to lazy load the UI components when they're
    * in the viewpoint.
    */
   items$: Observable<Observable<Product>[]> = this.componentData$.pipe(
+    filter((data) => Boolean(data.productCodes)),
```

Comment:
Maybe we should cover this canse in the map bellow, falbacking to `[]` ? To make it more bulletproof, for cases, when components will loose thier data because of some restrictions for new context (but for some reason still be visible?)

Reply:
good point, will do